QMLComponents/boundcontrols/boundcontrolbase.cpp

```diff
@@ -117,44 +117,49 @@ std::string BoundControlBase::getName() const
 	return fq(_control->name());
 }
 
-void BoundControlBase::_readTableValue(const Json::Value &value, const std::string& key, bool hasMultipleTerms, Terms& terms, ListModel::RowControlsValues& allControlValues, const Terms& sourceTerms)
+void BoundControlBase::_readTableValue(const Json::Value &value, const std::string& keyValue, const std::string& keyLabel, bool hasMultipleTerms, Terms& terms, ListModel::RowControlsValues& allControlValues, const Terms& sourceTerms)
 {
 	for (const Json::Value& row : value)
 	{
-		Json::Value keyValue = row[key];
-
-		Term term = Term::readTerm(keyValue);
+		Term term = Term::readTerm(row[keyValue]);
+		if (!keyLabel.empty() && row.isMember(keyLabel) && row[keyLabel].isString())
+			term.setLabel(tq(row[keyLabel].asString()));
```

Comment:
This should be done in Term!

Also instead of Term::readTerm I suggest just making proper constructors for Term.
Then you can give json:Value or string or double.
The term class can then contain the logic for getting labels out.

Why is all that stuff done in this file?

Reply:
Yes I'll make a Term constructor with a json argument.